src/main/java/com/cloudbees/plugins/deployer/resolvers/PromotionCapabilitiesResolver.java

```diff
@@ -119,7 +119,10 @@ protected boolean knows(@NonNull Class<? extends AbstractProject> jobType) {
      */
     @Override
     public Run<?, ?> getLastSuccessfulBuild(AbstractProject<?, ?> project) {
-        return CapabilitiesResolver.getLastDeployableBuild(PromotionProcess.class.cast(project).getRootProject());
+        if (project != null) {
+            return CapabilitiesResolver.getLastDeployableBuild(PromotionProcess.class.cast(project).getRootProject());
```

Comment:
Pointless use of `Class.cast` (should never be used with a constant receiver). Just

```java
(PromotionProcess) project
```